src/main/java/net/sf/mzmine/modules/peaklistmethods/filtering/duplicatefilter/DuplicateFilterTask.java

```diff
@@ -229,25 +230,27 @@ private PeakList filterDuplicatePeakListRows(final PeakList origPeakList, final
    * @param firstRow
    * @param secondRow
    */
-  private void createConsensusRow(RawDataFile[] rawFiles, PeakListRow firstRow,
+  private void createConsensusFirstRow(RawDataFile[] rawFiles, PeakListRow firstRow,
       PeakListRow secondRow) {
     for (RawDataFile raw : rawFiles) {
       Feature f2 = secondRow.getPeak(raw);
-      if (f2 != null) {
-        if (f2.getFeatureStatus().equals(FeatureStatus.DETECTED)) {
+      if (f2 == null)
+        continue;
+
+      switch (f2.getFeatureStatus()) {

```

Comment:
I have changed it to `if(f2==null) continue;` 
Although I am not sure which style is more readable.  Still needs to loop through all raw data files. 

And I have added the switch block.